PilotDataPlatform/bff-web
# PILOT-1589: Update project file activity statistics endpoint to be limited by role

## Summary

- Updated project file activity statistics endpoint to be limited by role, as opposed to solely by username: 
`POST /v1/project/{project_code}/activity`. 

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1589

## Type of Change

Please delete options that are not relevant.


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Login to Pilot (each time with different role - project admin, platform admin, collaborator, contributor)
2. File actions returned for heatmap should now be based on user role (e.g., contributor and collaborator should only be able to view their own files, while platform/project admin should be able to view all file actions for a given project)
